pkg/collector/builder.go

```diff
@@ -0,0 +1,83 @@
+/*
```

Comment:
Not sure `builder.go` is the right name here. Maybe `metric_family.go` instead as it is mostly concerned of groups of metrics?

Reply:
Yes, you are right, should we move these to the metrics pkg instead to `metric_family.go`?

Reply:
That is a great idea! Some of these could even be methods on the `Family` struct.

Reply:
@mxinden which ones of the below were you thinking of binding to the `Family` struct? As all of them accept `[]metrics.FamilyGenerator` and binding them to a single instance of the `FamilyGenerator` we would still need the functions that loop through the slices. 

Also was thinking of moving it to `pkg/metrics/family.go`, sgty?

Reply:
> Keeping the looping logic in ExtractMetricFamilyHeaders.

Where would that live?

Reply:
Maybe alongside the `FamilyGenerator` struct inside `pkg/metrics/generator.go`?